src/BenchmarkDotNet/Horology/TimeInterval.cs

```diff
@@ -1,15 +1,22 @@
-﻿using BenchmarkDotNet.Extensions;
+﻿using System.Text;
+using BenchmarkDotNet.Extensions;
 using JetBrains.Annotations;
 
 namespace BenchmarkDotNet.Horology
 {
     public struct TimeInterval
     {
+        private readonly Encoding encoding;
```

Comment:
We shouldn't keep encoding in `TimeInterval` because we need this value only for formatting. The `ToString()` implementation can use ASCII by default. Also, it makes sense to introduce additional `ToTimeStr(Encoding encoding)` for advanced formatting.